### PR TITLE
refactor: 171 panorama example refactor

### LINKED ISSUE #171: Panorama
Improve documentation of [panorama](https://registry.terraform.io/modules/PaloAltoNetworks/vmseries-modules/google/latest/examples/panorama) example

 - [ ] Provide code snippets to  generate and update the SSH key on the inputs.

### PR DESCRIPTION
**Pull request by horiagunica**
## Description

- Panorama examples refactor
- This PR closes #171 

## Motivation and Context

- Panorama example flexibility and documentation

## Types of changes

- Refactor

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
       - name: Set up Terraform
         uses: hashicorp/setup-terraform@v1
         with:
-          terraform_version: 0.15.3
+          terraform_version: 1.0.0 # Bumped from 0.15.3 (28/03/2023s)
```
Comment (michalbil) [inline .github/workflows/ci.yml:61]:
```suggestion
          terraform_version: 1.0.0
```

File: examples/panorama/README.md [hunk 2]
```diff
@@ -1,22 +1,91 @@
-# Palo Alto Networks Panorama Module Example
+# Palo Alto Panorama deployment example
 
->Panorama is a centralized management system that provides global visibility and control over multiple Palo Alto Networks next generation firewalls through an easy to use web-based interface. Panorama enables administrators to view aggregate or device-specific application, user, and content data and manage multiple Palo Alto Networks firewalls—all from a central location.
+## Overview
 
-This folder shows an example of Terraform code that helps to deploy Panorama in the Google Cloud Platform.
+The scope of this code is to successfuly deploy one or more panorama instances in a single project and region in Google Cloud.
```
Comment (michalbil) [inline examples/panorama/README.md:5]:
```suggestion
The scope of this code is to deploy one or more Panorama instances in a single project and region in Google Cloud.
```

File: examples/panorama/README.md [hunk 3]
```diff
@@ -1,22 +1,91 @@
...
 
-## Usage
+Important information :
 
-Create a `terraform.tfvars` file and copy the content of `example.tfvars` into it, adjust if needed.
+ - The code builds a single region topology for Panorama
```
Comment (michalbil) [inline examples/panorama/README.md:9]: This is already stated in the line above.

File: examples/panorama/variables.tf [hunk 4]
```diff
@@ -1,18 +1,105 @@
...
 # VPC
-variable "vpc_name" {}
-variable "subnet_name" {}
-variable "cidr" {}
-variable "allowed_sources" {}
+variable "vpcs" {
+  description = <<-EOF
+    A map containing each network setting:
+
+    Available options :
```
Comment (michalbil) [inline examples/panorama/variables.tf:18]: For this kind of details let's maybe add a link directly to module input description?

**Review by michalbil — CHANGES_REQUESTED** [review]

**Comment by migara** [comment]
@horiagunica let's add the topology image directly onto the README. Just copy and paste the image onto your markdown

File: examples/panorama/main.tf [hunk 5]
```diff
@@ -1,36 +1,41 @@
...
+  source  = "PaloAltoNetworks/vmseries-modules/google//modules/panorama"
+  version = "0.5.1"
 
-  name              = var.panorama_name
+  for_each = var.panoramas
+
+  name              = each.value.panorama_name
   project           = var.project
   region            = var.region
   zone              = data.google_compute_zones.this.names[0]
```
[the remark this answers is not in the document]
Comment (horiagunica) [reply examples/panorama/main.tf]: Added the zone as input in tfvars.

File: examples/panorama/main.tf [hunk 6]
```diff
@@ -1,36 +1,41 @@
...
-  attach_public_ip  = var.attach_public_ip
-  log_disks         = var.log_disks
-}
+  panorama_version  = each.value.panorama_version
+  ssh_keys          = each.value.ssh_keys
+  subnet            = each.value.panorama_subnet
+  private_static_ip = each.value.private_static_ip
+  attach_public_ip  = each.value.attach_public_ip
+  log_disks         = each.value.log_disks
+  depends_on        = [module.vpc]
```
[the remark this answers is not in the document]
Comment (horiagunica) [reply examples/panorama/main.tf:26]: Reffered VPC module

File: examples/panorama/README.md [hunk 7]
```diff
@@ -1,28 +1,92 @@
...
+ - A panorama instance with a Public IP address attached to the created vpc network and subnetwork
+ - Firewall rules that allow access to the panorama management interface
+
+![panorama-topology](https://user-images.githubusercontent.com/43091730/230029801-3acea62e-aa3d-46f3-b638-6b09bf5ef35e.png)
+
+## Prerequisites
+
+1. Prepare [panorama license](https://support.paloaltonetworks.com/)
+
+2. Configure the terraform [google provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/getting_started#configuring-the-provider)
```
[the remark this answers is not in the document]
Comment (horiagunica) [reply examples/panorama/README.md]: You are right, applied!

File: examples/panorama/variables.tf [hunk 8]
```diff
@@ -1,18 +1,69 @@
...
+}
+variable "region" {
+  description = "The region into which to deploy the infrastructure in to"
+  type        = string
+  default     = "us-central1"
+}
+variable "name_prefix" {
+  description = "A string to prefix resource namings"
+  type        = string
+  default     = "example-"
```
[the remark this answers is not in the document]
Comment (horiagunica) [reply examples/panorama/variables.tf:15]: Defaulted to empty string.

**Review by migara — APPROVED** [review]
Great work!

File: examples/panorama/outputs.tf [hunk 9]
```diff
@@ -1,9 +1,9 @@
 output "panorama_private_ip" {
```
Comment (michalbil) [inline examples/panorama/outputs.tf:1]:
```suggestion
output "panorama_private_ips" {
```

File: examples/panorama/outputs.tf [hunk 10]
```diff
@@ -1,9 +1,9 @@
...
   description = "Private IP address of the Panorama instance."
-  value       = module.panorama.panorama_private_ip
+  value       = { for k, v in module.panorama : k => v.panorama_private_ip }
 }
 
 output "panorama_public_ip" {
```
Comment (michalbil) [inline examples/panorama/outputs.tf:6]:
```suggestion
output "panorama_public_ips" {
```

File: examples/panorama/variables.tf [hunk 11]
```diff
@@ -1,18 +1,69 @@
...
+  description = <<-EOF
+    A map containing each network setting.
+
+    Example of variable deployment :
+
+    ```
+    vpcs = {
+      "panorama-vpc" = {
+        vpc_name          = "panorama-vpc"
+        subnet_name       = "example-panorama-subnet"
```
Comment (michalbil) [inline examples/panorama/variables.tf:29]:
```suggestion
        subnet_name       = "panorama-subnet"
```

File: examples/panorama/variables.tf [hunk 12]
```diff
@@ -1,18 +1,69 @@
...
+    A map containing each panorama setting.
+
+    Example of variable deployment :
+
+    ```
+    panoramas = {
+      "panorama-01" = {
+        panorama_name     = "panorama-01"
+        panorama_vpc      = "panorama-vpc"
+        panorama_subnet   = "example-panorama-subnet"
```
Comment (michalbil) [inline examples/panorama/variables.tf:56]:
```suggestion
        panorama_subnet   = "panorama-subnet"
```

**Review by michalbil — APPROVED** [review]

**Pull request merged**